clippy_lints/src/inherent_to_string.rs

```diff
@@ -0,0 +1,148 @@
+use if_chain::if_chain;
+use rustc::hir::{ImplItem, ImplItemKind};
+use rustc::lint::{LateContext, LateLintPass, LintArray, LintPass};
+use rustc::{declare_lint_pass, declare_tool_lint};
+
+use crate::utils::{
+    get_trait_def_id, implements_trait, in_macro_or_desugar, match_type, paths, return_ty, span_help_and_lint,
+    trait_ref_of_method, walk_ptrs_ty,
+};
+
+declare_clippy_lint! {
+    /// **What id does:** Checks for the definition of inherent methods with a signature of `to_string(&self) -> String`.
+    ///
+    /// **Why is this bad?** This method is also implicitly defined if a type implements the `Display` trait. As the functionality of `Display` is much more versatile, it should be preferred.
+    ///
+    /// **Known problems:** None
+    ///
+    /// ** Example:**
+    ///
+    /// ```rust,ignore
+    /// // Bad
+    /// pub struct A;
+    ///
+    /// impl A {
+    ///     pub fn to_string(&self) -> String {
+    ///         "I am A".to_string()
+    ///     }
+    /// }
+    /// // Good
```

Comment:
Splitting up the good and bad example should make this examples testable.

```suggestion
    /// ```
    ///
    /// ```rust
    /// // Good
```